src/HDWallet.cpp

```
@@ -241,64 +204,75 @@ std::string HDWallet<seedSize>::getExtendedPublicKeyAccount(TWPurpose purpose, T
 
     const auto curve = TWCoinTypeCurve(coin);
     const auto path = TW::derivationPath(coin, derivation);
-    auto derivationPath = DerivationPath({DerivationPathIndex(purpose, true), DerivationPathIndex(path.coin(), true)});
-    auto node = getNode(*this, curve, derivationPath);
-    auto fingerprintValue = fingerprint(&node, publicKeyHasher(coin));
-    hdnode_private_ckd(&node, account + 0x80000000);
-    hdnode_fill_public_key(&node);
-    return serialize(&node, fingerprintValue, version, true, base58Hasher(coin));
+    auto derivationPath = DerivationPath({DerivationPathIndex(purpose, true), DerivationPathIndex(path.coin(), true), DerivationPathIndex(account, true)});
+    auto node = getNode(*this, curve, derivationPath, TW::publicKeyHasher(coin));
+    if (node == nullptr) {
+        return "";
```

Let's throw an exception?